runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/control/FnApiControlClient.java

```diff
@@ -158,6 +158,8 @@ public void onNext(BeamFnApi.InstructionResponse response) {
                       "Error received from SDK harness for instruction %s: %s",
                       response.getInstructionId(), response.getError())));
         }
+      } else {
+        LOG.warn("Dropped unknown InstructionResponse {}", response);
```

Comment:
Tiny style suggestion: move the error case to the top. Reduce one level of curly braces.

```java
if (responseFuture == null) {
  LOG.warn(...)
  return;
}
```